machine/plugins/base.py

```diff
@@ -336,6 +339,17 @@ def say(self, text: str,
         else:
             ephemeral_user = None
 
+        if self.channel is None:
```

Comment:
I don't think this will fully fix your problem. If the channel is not in the client's channel dict, then the property lookup `self.channel` will still throw an error. In order to fix that, you'd need to change L278 to do a `.get(...)` instead of a square bracket lookup.